src/routes/Dashboard/TaskViewEditModal.js

```diff
@@ -0,0 +1,308 @@
+import * as React from "react";
+
+import api from "../../api/axios";
+
+import Spinner from "../../components/Spinner";
+
+import * as Mui from "@mui/material";
+
+import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
+import { CircleOutlined } from "@mui/icons-material";
+
+import { ReactComponent as InboxIcon } from "../../icons/TaskEditModal/inbox.svg";
+import { ReactComponent as UserIcon } from "../../icons/TaskEditModal/user.svg";
+import { ReactComponent as DateIcon } from "../../icons/TaskEditModal/date.svg";
+import { ReactComponent as FlagIcon } from "../../icons/TaskEditModal/flag.svg";
+import { ReactComponent as PlusIcon } from "../../icons/TaskEditModal/plus.svg";
+import { ReactComponent as MoveUpIcon } from "../../icons/TaskEditModal/move_up.svg";
+import { ReactComponent as MoveDownIcon } from "../../icons/TaskEditModal/move_down.svg";
+import { ReactComponent as MenuIcon } from "../../icons/TaskEditModal/menu.svg";
+import { ReactComponent as CloseIcon } from "../../icons/TaskEditModal/close.svg";
+
+import { ReactComponent as PriorityIcon } from "../../icons/icon_priority.svg";
+
+export default function TaskViewEditModal({
+  open,
+  closeTaskModal,
+  selectedTaskId,
+}) {
+  const [task, setTask] = React.useState({});
+  const [loading, setLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    async function getTask() {
+      await getTaskFromApi(selectedTaskId);
+    }
+
+    getTask();
+  }, [selectedTaskId]);
+
+  const getTaskFromApi = async (taskId) => {
+    const { data } = await api.get(`/api/tasks/${taskId}`);
+
+    //get name of month from Date():https://stackoverflow.com/questions/1643320/get-month-name-from-date
+    const dueDate = new Date(data.due_date);
+    data.due_date_day = dueDate.getDay();
+    data.due_date_month_name = dueDate
+      .toLocaleString("default", { month: "long" })
+      .substring(0, 3);
+
+    setTask(data);
+    setLoading(false);
+  };
+
+  const close = () => closeTaskModal();
```

Comment:
I suggest we rename this function as 'handleClose'. Going forward, I suggest we use 'handle<Event>' naming convention for event handlers.

Reply:
Good suggestion